kobo/apps/openrosa/apps/logger/xform_fs.py

```diff
@@ -1,35 +1,61 @@
 # coding: utf-8
-import os
 import glob
+import os
 import re
 
+from lxml import etree
+
 
 class XFormInstanceFS:
     def __init__(self, filepath):
         self.path = filepath
         self.directory, self.filename = os.path.split(self.path)
-        self.xform_id = re.sub(".xml", "", self.filename)
+        self.xform_id = re.sub('.xml', '', self.filename)
+
+    @property
+    def attachments(self):
+        if not hasattr(self, '_attachments'):
+            dir = os.path.join(self.directory)
+            self._attachments = [
+                entry
+                for entry in os.scandir(dir)
+                if entry.is_file()
+                and entry.path != self.path
+                and entry.name in self.mentioned_in_xml
+            ]
+        return self._attachments
 
     @property
     def photos(self):
         if not hasattr(self, '_photos'):
-            available_photos = glob.glob(os.path.join(self.directory, "*.jpg"))
+            available_photos = glob.glob(os.path.join(self.directory, '*.jpg'))
             self._photos = []
             for photo_path in available_photos:
                 _pdir, photo = os.path.split(photo_path)
                 if self.xml.find(photo) > 0:
                     self._photos.append(photo_path)
         return self._photos
 
+    @property
+    def mentioned_in_xml(self):
+        if not hasattr(self, '_mentioned_in_xml'):
+            parser = etree.XMLParser()
+            root = etree.fromstring(self.xml, parser=parser)
+            namespaces = root.nsmap
+            self._mentioned_in_xml = set(
+                root.xpath(
+                    '//*/text()',
```

Comment:
I had this as `//*/[@type="file"]/text()`, but that was too strict for `test_importing_b1_and_b2`:

```xml
<?xml version='1.0' ?>
<tutorial id="tutorial">
	<name>A</name>
	<age>1</age>
	<picture>1330640038667.jpg</picture>
	<has_children>0</has_children>
	<gps />
	<web_browsers>ie</web_browsers>
	<meta>
		<instanceID>uuid:435f173c688e482486a48661700467gh</instanceID>
	</meta>
</tutorial>
```

Reply:
IMO we should match the behavior when handling single submissions and accept *all* attached files regardless of the XML submission content. In the future, we could be more discerning for all submission handling, but I want to minimize the discrepancies between this bulk code and the standard single-submission logic. (The rest of the PR does help achieve that.)

Additionally, we have to watch out for things like background audio and audit logs, which don't appear the same way in the submission XML as regular audio/video/file questions. Here's an example (probably not the best) of us having to deal with that: https://github.com/kobotoolbox/kobocat/pull/905.

@noliveleger I'll give you the determination here.

Reply:
Resolved by removing `mentioned_in_xml` logic :ok_hand: